server/src/routes/api/v1/userActivityRouter.js

```diff
@@ -0,0 +1,28 @@
+import express from "express"
+
+import { User } from "../../../models/index.js"
+
+const userActivityRouter = new express.Router({ mergeParams: true })
+
+userActivityRouter.get("/", async (req, res) => {
+    const userData = req.user
+
+    try {
+        const user = await User.query().findById(userData.id)
+        user.salads = await user.$relatedQuery("salads")
+        user.reviews = await user.$relatedQuery("reviews")
+        const userSaladsSorted = user.salads.sort((b,a) => {
+            return Date.parse(a.createdAt) - Date.parse(b.createdAt)
+        })
+        const sortedSaladsWithUser = await Promise.all(userSaladsSorted.map(async(salad) => {
+            salad.user = await salad.$relatedQuery("user")
+            return salad
+        }))
```

Comment:
why are we querying for the user of each salad? based on line 12, aren't the salads created by this current user?